[git] Allow pre-commit hook to feed from staged ChangeLogs

philn opened the pull request:
#### d451029f5e9461d561da1b09ceab6d793e4c9469
<pre>
[git] Allow pre-commit hook to feed from staged ChangeLogs
<a href="https://bugs.webkit.org/show_bug.cgi?id=239419">https://bugs.webkit.org/show_bug.cgi?id=239419</a>

Patch by Philippe Normand &lt;pnormand@igalia.com &gt; on 2022-04-25
Reviewed by Jonathan Bedard.

In workflows where prepare-ChangeLog is executed manually before staging and committing
changes the prepare-commit-msg hook can now read the curated ChangeLog from the git stage
and present it in the editor for validation. In case no ChangeLog was staged, the hook will
generate a message itself.

* Scripts/hooks/prepare-commit-msg:

Canonical link: <a href="https://commits.webkit.org/249941@main">https://commits.webkit.org/249941@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293318">https://svn.webkit.org/repository/webkit/trunk@293318</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

philn commented on Tools/Scripts/hooks/prepare-commit-msg:
```diff
@@ -3,13 +3,39 @@
 
+def message_from_staged_changelogs():
+    try:
+        diff = subprocess.check_output(['git', 'diff', '--staged'])
```
Ah, this git should be `{{ git }}` I suspect.

JonWBedard commented on Tools/Scripts/hooks/prepare-commit-msg:
```diff
@@ -3,13 +3,39 @@
 import os
 import subprocess
 import sys
+from webkitpy.common.checkout.diff_parser import DiffParser
```
This isn't going to work reliably.

You'll need to make sure `SCRIPTS` is on the Python path before importing because the user might actually be setting up a different repository with WebKit's tooling. That's why the old line 18/new line 44 invokes `prepare-ChangeLog` from `SCRIPTS` specifically.

JonWBedard reviewed (changes requested):
You'll need to change the Python path to make sure this continues to work with repositories other than WebKit, but other than that, I think this is a good idea.

zdobersek committed:
[RISCV64] Enable Baseline JIT
https://bugs.webkit.org/show_bug.cgi?id=239708

Reviewed by Yusuke Suzuki.

.:

Turn on the CMake defaults for ENABLE(JIT) and ENABLE(FTL_JIT) options
for RISCV64. These are enabled together to avoid having to worry about
a 64-bit platform that would want to allow compiling with Baseline JIT
enabled and FTL JIT disabled at the same time. But because FTL JIT
support for RISCV64 isn't complete yet, it's still disabled at runtime.

* Source/cmake/WebKitFeatures.cmake:

Source/JavaScriptCore:

* runtime/Options.cpp:
(JSC::Options::recomputeDependentOptions):
Disable FTL JIT at runtime under CPU(RISCV64) since the support is still
incomplete. This is necessary due to Baseline JIT and FTL JIT still
being enabled at build-time on this platform for pure convenience.

Canonical link: https://commits.webkit.org/249939@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293316 268f45cc-cd09-0410-ab3c-d52691b4dbfc

kkinnunen-apple committed:
REGRESSION (Safari 15.4) Performance regression after uploading WebGL buffers
https://bugs.webkit.org/show_bug.cgi?id=239015

Patch by Kimmo Kinnunen <kkinnunen@apple.com> on 2022-04-25
Reviewed by Dean Jackson.

BufferMtl::getRestartIndices() would recalculate restart indices on each
draw because RestartRangeCache::operator bool() would return m_isDirty
instead of the intended !m_isDirty.

Use std::optional to hold the RestartRangeCache instead of
maintaining m_isDirty. This allows the type system to contribute to the
correctness of the code. Dirty restart range cache is not useful and takes
up memory.

Re-apply hunks from r281550 "ANGLE Metal index buffer left mapped when building primitive restart ranges".

These were overwritten in r286638 "Roll ANGLE to include upstreamed Metal backend".

* src/libANGLE/renderer/metal/BufferMtl.h:
(rx::BufferMtl::RestartRangeCache::RestartRangeCache):
* src/libANGLE/renderer/metal/BufferMtl.mm:
(rx::BufferMtl::markConversionBuffersDirty):
(rx::BufferMtl::clearConversionBuffers):
(rx::BufferMtl::getRestartIndices):

Canonical link: https://commits.webkit.org/249940@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293317 268f45cc-cd09-0410-ab3c-d52691b4dbfc

philn committed:
[git] Allow pre-commit hook to feed from staged ChangeLogs
https://bugs.webkit.org/show_bug.cgi?id=239419

Patch by Philippe Normand <pnormand@igalia.com> on 2022-04-25
Reviewed by Jonathan Bedard.

In workflows where prepare-ChangeLog is executed manually before staging and committing
changes the prepare-commit-msg hook can now read the curated ChangeLog from the git stage
and present it in the editor for validation. In case no ChangeLog was staged, the hook will
generate a message itself.

* Scripts/hooks/prepare-commit-msg:

Canonical link: https://commits.webkit.org/249941@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293318 268f45cc-cd09-0410-ab3c-d52691b4dbfc
